bebraven/platform
# When creating an account, handle errors better.

### Description
Task: https://app.asana.com/0/1174274412967132/1199986710909292

I don't know how this ever worked since [this change](https://github.com/bebraven/platform/commit/445429fa5560785f35eb71d6d5e8416faa01900c) 10 months ago?!?

Maybe a recent upgrade with ruby/rails/devise changed behavior? Either way, when you put in a password that is too short or mismatched, `resource.errors.first` is an `ActiveModel::Error`.

### Test Plan
1. Get you a fully confirmed Participant in the Salesforce instance you're configured against.
2. Hit https://platformweb/users/sign_up?u=<SF_Contact_ID>
3. Try out mismatched passwords and too short passwords
4. Then use a valid password and ensure that you get the confirmation email (at least in the logs).
5. Make sure that the `canvas_user_id` is set on the platform `User` model and that Canvas user was actually created over in Canvas for the "Highlander Accelerator Course ID" set in Salesforce on the Program that the user is a Participant in.

### Screenshots (if appropriate):
Password too short error
![image](https://user-images.githubusercontent.com/5596986/112375985-49821680-8cba-11eb-8627-49ad31653d1b.png)

Passwords don't match error
![image](https://user-images.githubusercontent.com/5596986/112376033-57d03280-8cba-11eb-9aac-cd0679917f27.png)


### Types of changes




### Checklist:
